### DISCUSSION TITLE
Handling multiple specs

### DISCUSSION THREAD

**Discussion by mrchimp** [scalar/scalar#797] [Q&A]
I have several specs defined in separate JSON files and I'm trying to find a good way to access them through the same UI.

I have tried adding a wrapper around `<ApiReference>` that allows a user to select the appropriate spec and then updating the `configuration` prop. I ran into problems where it either didn't notice that the configuration and therefore nothing changed. Or both configurations became merged together and the paths and models for both were displayed.

It would be nice to have the option to either:

1. Pass all of the configurations to `<ApiReference>` and have it provide a selector to allow switching between them
2. Allow updating the configuration in real-time. I presume this must be possible as that's how the Swagger Editor appears to work but I couldn't make it happen, so some guidance on this would be welcome.

Also, thanks for your work on Scalar. I've raised a few issues but it's looking great already.

**Comment by hanspagel** [comment]
Thanks for bringing this up! I think adding a switcher is out of scope for the project, but it should be possible to add this in your project.

How did you update the configuration? I think you’d need to make the configuration a `reactive()` object, and use `Object.assign(configuration, newConfiguration)` to trigger an update.

**Reply by mrchimp** [reply]
I just gave it another shot. I'm doing something like this:

```javascript
const configuration = reactive({
  metaData: {
    title: "My docs",
    description: "My docs",
  },
  isEditable: false,
  spec: {
    content: null,
  }
});

async function loadContent(url) {
  const response = await fetch(url, {
    method: "GET",
  });

  const content = await response.json();

  // ...do some content parsing/updating...

  const newConfig = {
    ...configuration,
    spec: {
      content,
    },
  };

  Object.assign(configuration, newConfig);
}
```

Then when a spec is selected I call `loadContent` with the appropriate URL. It seems to work in as much as when I inspect `<ApiReference>` in dev tools I can see the updated content, but nothing changes on the page.

**Reply by hanspagel** [reply]
Are you using Vue 3.4? I’ve had an issue with that approach in Vue 3.4, but not in Vue 3.3. Didn’t had time to check what changed though.

**Reply by mrchimp** [reply]
Yeah, looks like it's a Vue 3.4 issue. Rolling back to 3.3 makes it work.

Is there a different approach that works with 3.4?
